backend/src/controllers/mining.controller.ts

```diff
@@ -273,30 +278,55 @@ export default function initializeMiningController(
       return res.status(201).send({ error: null, data: miningTask });
     },
 
-    startMiningFile(req: Request, res: Response) {
+    async startMiningFile(req: Request, res: Response, next: NextFunction) {
       const user = res.locals.user as User;
 
       const {
         name,
         contacts
       }: {
         name: string;
-        contacts: Record<string, string>[];
+        contacts: Partial<Contact[]>;
       } = req.body;
 
-      return res
-        .status(200)
-        .send({ error: null, name, contacts, userId: user.id });
+      try {
+        validateFileContactsData(contacts);
```

Comment:
You should expect output from this function and if validation false return 400 to the user. Otherwise it will return 500 which is unclear.